src/_singletons/ContractDetailsSingleton.js

```diff
@@ -0,0 +1,29 @@
+import React, { PropTypes, Component } from 'react';
+import { connect } from 'react-redux';
+import Modal from '../containers/Modal';
+import ContractDetailsContainer from '../contract-details/ContractDetailsContainer';
+import ContractDetailsSingletonSelector from './ContractDetailsSingletonSelector';
+
+@connect(ContractDetailsSingletonSelector)
+export default class ContractDetailsSingleton extends Component {
```

Comment:
Possibly Container is better name than Singleton.
Singleton is describing a bit of its purpose, but not rally the pattern, as I can certainly create more than one of it ;)